### DISCUSSION TITLE
RateLimiting and ASP.Net

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#67910] [General]
I just saw System.Threading.RateLimiting. 

Are there any components or middleware's that allow a json API to emit rats limiting response headers?

In a similar vein, are their any http handlers that could pick up the headers and automatically throttle so that the http requests don't overload the server?

**Comment by dave-yotta** [comment]
I've used https://github.com/stefanprodan/AspNetCoreRateLimit with success.
